procurement_batch_generator/wizard/procurement_batch_generator.py

```diff
@@ -63,23 +49,23 @@ def _default_lines(self):
     def validate(self):
         self.ensure_one()
         wiz = self[0]
-        assert wiz.line_ids, 'wizard must have some lines'
-        new_po_ids = []
+        if not wiz.line_ids:
+            raise UserError(_('There are no lines!'))
+        poo = self.env['procurement.order']
+        procs = poo
+        prec = self.env['decimal.precision'].precision_get(
+            'Product Unit of Measure')
         for line in wiz.line_ids:
-            if not line.procurement_qty:
+            if float_is_zero(line.procurement_qty, precision_digits=prec):
                 continue
-            procurement = self.env['procurement.order'].create(
-                line._prepare_procurement_order())
-            new_po_ids.append(procurement.id)
-        if not new_po_ids:
-            raise Warning(_('All requested quantities are null.'))
-        self.pool['procurement.order'].signal_workflow(
-            self._cr, self._uid, new_po_ids, 'button_confirm')
-        self.pool['procurement.order'].run(
-            self._cr, self._uid, new_po_ids, context=self.env.context)
+            proc = poo.create(line._prepare_procurement_order())
+            procs += proc
```

Comment:
I think it would be nice to have these created with a procurement group. When you receive in 2 steps, this allows to have the input -> stock pickings grouped by procurement group which is nice.